# AnomieBOT

https://en.wikipedia.org/wiki/User:AnomieBOT

Practically endless list of tasks, operating since at least 2008, has 5 different accounts to do edits


In [1]:
%reload_ext autoreload
%autoreload 2
%matplotlib inline

In [2]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

In [3]:
from datetime import datetime

In [4]:
df = pd.read_csv('revs_scored_jan.tsv', sep='\t', header=0)

In [5]:
days = df.rev_timestamp.map(lambda ts: datetime.utcfromtimestamp(ts).day)
df['day'] = days

In [6]:
sdf_anomiebot = df[df.day<22]
sdf_anomiebot = sdf_anomiebot[sdf_anomiebot.user_text == "AnomieBOT"]

sdf_reverted = sdf_anomiebot[sdf_anomiebot.is_reverted] 
sdf_reverted = sdf_reverted[sdf_reverted.seconds_to_revert.astype('str').astype('int')<86400]

In [14]:
sdf1 = sdf_reverted.loc[sdf_reverted.is_reverted, ["rev_id", "user_text", "revert_id", "page_namespace"]]
sdf1.revert_id = sdf1.revert_id.astype('int')

In [23]:
sdf2 = df[df.user_is_bot == False]
sdf2 = sdf2[sdf2.is_revert]
sdf2['revert_set_size'] = sdf2.revert_set_size.astype('int')
sdf2 = sdf2[["rev_id", 'user_text', 'revert_set_size']]

In [24]:
reverts_by_human = pd.merge(sdf1, sdf2, 
             how='inner', 
             left_on='revert_id', 
             right_on='rev_id', 
             suffixes=('', '_reverter')
)

In [25]:
reverts_by_human0 = reverts_by_human[reverts_by_human.page_namespace == 0]

In [26]:
# summary of variables
#   sdf_anomiebot = all edits by anomiebot w/in time frame
#   sdf_reverted = all reverted edits by anomiebot w/in time frame
#   reverts_by_human = all edits of AnomieBOT's that were reverted by a human
#   reverts_by_human0 = all edits by anomiebot in namespace 0 reverted by a human

## sample of edits reverted by humans in namespace 0

In [27]:
len(reverts_by_human0)

214

In [28]:
reverts_by_human0.sample(n=20, random_state=1).reset_index()

index     rev_id  user_text  revert_id  page_namespace  rev_id_reverter  \
0     120  877844261  AnomieBOT  877856433               0        877856433   
1     222  879465056  AnomieBOT  879584988               0        879584988   
2     118  877789628  AnomieBOT  877792114               0        877792114   
3     119  877808458  AnomieBOT  877838653               0        877838653   
4      54  877003846  AnomieBOT  877036217               0        877036217   
5      35  876751654  AnomieBOT  876878685               0        876878685   
6      64  877147243  AnomieBOT  877155583               0        877155583   
7     165  878689146  AnomieBOT  878825529               0        878825529   
8       4  876274193  AnomieBOT  876315021               0        876315021   
9      33  876737595  AnomieBOT  876819326               0        876819326   
10    131  878030255  AnomieBOT  878043772               0        878043772   
11     42  876865419  AnomieBOT  876866428               0        876866428   
12    104  877638107  AnomieBOT  877736405               0        877736405   
13     77  877335896  AnomieBOT  877389055               0        877389055   
14    175  878860832  AnomieBOT  878926472               0        878926472   
15    124  877899322  AnomieBOT  878020595               0        878020595   
16    105  877655493  AnomieBOT  877730872               0        877730872   
17    108  877687214  AnomieBOT  877694401               0        877694401   
18    164  878685382  AnomieBOT  878820126               0        878820126   
19     41  876865186  AnomieBOT  876870728               0        876870728   

   user_text_reverter  revert_set_size  
0           IndianBio               13  
1            Mcbjmund                2  
2          SlashFox14                3  
3           SmartTalk                4  
4               NPguy                2  
5      108.51.174.100                4  
6           Dlthewave                4  
7              JJMC89                2  
8               Drdpw                3  
9           Oknazevad                3  
10           Feminist                2  
11             Bilorv                8  
12             BD2412                2  
13            Lugnuts                2  
14             C.Fred                1  
15     Rohitmishra183                3  
16           Onel5969                1  
17       David Gerard                2  
18        Favre1fan93                4  
19      Adoring nanny                2

0. 877844261 comment: dating maintenance tags {{Cn}}, has minor flag. added date to {{cn}} template. original edit at 1am. reverting user says "revert to last good version before the IP crap started." 3am. also gets rid of other intervening edits. apparently cn template is the same as citation needed template, just abbreviated. ***human conflict***

1. 879465056 comment: rescuing orphaned refs. adds lots of info to a ref which only has a name, including web url, names, title, date and accessdate. reverting edit 16 hours later removes only AnomieBOT's additions. comment: this page is for the 2019 F3 Asian Championship. Ghiretti is confirmed to run in this series with Hitech, the regular series and the winter series have separate entry lists. I believe that this is ***human error***, because the ref name "Hitech" is left undefined.

2. 877789628 comment: dating maintenance tags: {{fact}}. AnomieBOT adds date to fact tag. 20 mins later, a user comes along and deletes that whole line, including tag, saying that there is no official announcement for the info yet. ***human conflict***

3. 877808458 comment: dating maintenance tags: {{Citation needed}}. AnomieBOT adds date to citation needed tag. 5 hours later, a user comes along and deletes the entire sentence, including the tag, basically saying that they couldn't find any supporting info. ***human conflict***

4. 877003846 comment: Dating maintenance tags: {{Use dmy dates}}. Anomie BOT changes tag date from April 2019 to January 2019. another user 5 hours later leaves no edit summary, but changes Anomie's tag to April 2015 and changes another date in the article to 2015. I'm guessing Anomie changed the date because it hadn't happened yet, and the other user used the wrong date in the first place. it looks like this template is used to tell editors that dates in the article must have a specific format, and the date is used to show editors when the dates were last checked for the correct format by a human or a bot. I think the human user may have been incorrect about the usage of this template and shouldn't have made that change. ***human error?***

5. 876751654 comment: Dating maintenance tags: {{Cn}}. AnomieBOT adds date to tag. 20 hours later, user says: removing unsourced information about palms which is simply not true. ***human conflict***

6. 877147243 comment: Rescuing orphaned refs. AnomieBOT adds lots of info to ref that only had name, including url and title, and access-date. reverting edit an hour later says: as discussed on the talk page, these sources do not discuss social media addiction and are therefore WP:SYN. this edit removes multiple paragraphs, including the citation edited by the bot. ***human conflict***

7. 878689146 comment: fixing reference errors. AnomieBOT removes "\<ref>\</ref>" from the page. reverting edit 20 hours later is completely unrelated, made on line 66 where Anomie's edit was on line 1. routine work.

8. 876274193 comment: dating maintenance tags: {{2l}}. AnomieBOT adds date to 2l template. reverting edit 7 hours later after intervening edits, comment says: "reverted to revision 875227896 by Drdpw (talk): All names transferred. (TW)". 2l template appears to be short for too long, same as the {{VeryLong}} template. I think the reverting user disagreed that the template and various other edits were necessary, as the revert target was to the edit before the 2l template was added. ***human conflict***

9. 876737595 comment: dating maintenance tags: {{Cn}} {{Or}}. the bot adds dates to the two tags, which are in the same paragraph. 14 hours later, a user removes the whole paragraph with the comment: remove conspiratorial BS as I have before. ***human conflict***

10. 878030255 comment: fixing reference errors and rescuing orphaned refs ("nvpi" from rev 877888036; "brit" from rev 877888036). the bot adds info to 2 refs that only have a name. it also adds quotation marks to the names. 2 hours later, same user that made previous edit reverts to revision 877888036 without leaving a specific comment. the edit appears to remove both lines that AnomieBOT edited, among several other things. 

11. 876865419 comment: fixing reference errors. bot removes empty ref template (\<ref>\</ref>). reverting comment: both of these links are already in the article. reverting edit 10 mins later removes two lines from the article, not including the one AnomieBOT edited. the two edits are really unrelated. routine work.

12. 877638107 comment: dating maintenance tags: {{incoming links}}. bot adds a date to the template. 17 hours later, user removes the whole template with comment: no longer incoming links. seems like routine work. 

13. 877335896 comment: dating maintenance tags: {{use dmy dates}}. bot changes date on template from October 2019 to January 2019, probably because it makes no sense to say October because it hadn't happened yet at the time. reverting edit 9 hours later changes date to September 2018. also changes things to say film is from 2018 rather than 2019 and other small edits. reverting comment simply reads 'lgv'. not sure what's going on here. however, AnomieBOT did not make an error and was functioning as intended. the first user made an error. ***human error***

14. 878860832 comment: rescuing orphaned refs ("rollingstone.com" from rev 878779391). bot adds link to ref with only name. reverting edit 8 hours later  comment says: reverted to revision 878779391 by C.Fred (talk): Rvv. this revert restores the ref that AnomieBOT rescued. ***human error***

15. 877899322 comment: fixing reference errors. bot removes two empty ref templates. the following edit removes some other nonsense syntax marks and changes a word, no ref summary. ***human error***

16. 877655493 comment: dating maintenance tags: {{lead too short}}. bot adds date to template. reverting comment: Reverted to revision 872196374 by Cat's Tuxedo (talk): Restore redirect - no indication of notability. ***human conflict*** error here - clearly reverted with other edits as well from page history https://en.wikipedia.org/w/index.php?title=The_Ren_%26_Stimpy_Show:_Time_Warp&action=history

17. 877687214 comment: dating maintenance tags: {{by whom?}}. bot adds date to template. reverting edit 2 hours later makes a LOT of changes, among which removes the entire template including the date added by the bot. reverting comment reads: rv weird grammar, links to dabs. routine work.

18. 878685382 comment: rescuing orphaned refs (MaxiRole from Cloak & Dagger). bot adds url, title, author, date, accessdate, and archiveurl to a ref with only name. reverting edit 3 hours later removes a section, including this ref. revert comment: reverted to revision 878085310 by Adamstom.97 (talk): no confirmation for Maxi for THIS season, nor that O'Reilly will go by Mayhem. ***human conflict***

19. 876865186 comment: rescuing orphaned refs ("everett" from rev 873063733). bot adds url, names, date, title to ref with only name. reverting edit 1 hour later adds in a paragraph above, including the missing ref info, and removes the bits that AnomieBOT added to the ref below. reverting comment: entirely appropriate for the lead to summarize the body. ***human conflict***


In [36]:
df_all_reverted = df[df.is_reverted]
df_all_reverted = df_all_reverted[df_all_reverted.user_is_bot == True]
len(df_all_reverted[(df_all_reverted.revert_set_size != '1') & (df_all_reverted.revert_set_size != 'None')]) / len(df_all_reverted)

0.5475533723786132

In [38]:
# percentage of reverted bot edits in namespace 0, that were reverted with at least one other edit
df_all_reverted0 = df_all_reverted[df_all_reverted.page_namespace ==0]
len(df_all_reverted0[(df_all_reverted0.revert_set_size != '1') & (df_all_reverted0.revert_set_size != 'None')]) / len(df_all_reverted0)

0.31217277486910994

In [29]:
len(reverts_by_human0[reverts_by_human0.revert_set_size > 1]) / len(reverts_by_human0) 

0.897196261682243

In [34]:
reverts_by_human.size

1624

In [35]:
reverts_by_human0.size

1498